tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/proxy_adv_multi_subnet_coex.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+# Copyright 2023 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+source $(dirname "${BASH_SOURCE[0]}")/../../_mesh_test.sh
+
+# Test Proxy advertisement Coex
+
+# This test verifies correct Proxy advertisement behavior for a device
+# where the Proxy adv requirements changes over time, both for single
+# and multiple subnets. The TX device is the DUT in this instance, while
+# the RX device scans and verifies that the correct proxy adv messages of
+# the different subnets is sent within the expected time delta.
+
+# Note 1: The maximum allowed timeslot for a subnet to
+# advertise proxy in this scenario is 10 seconds when there is more than
```

Comment:
"The maximum allowed timeslot for a subnet to advertise proxy in this scenario is 10 seconds **when there is more than one subnet that has active proxy adv work.**"

I thought that it was already stating that it applied to multi subnet scenario, but if it is unclear I can rephrase it. The main purpose of this note is to emphazies that in this particular test scenario the max TS is 10 seconds, so that the reader might get the timing/adv_count criteria in the test. For a test/usecase with more than 6 subnets, with active proxy adv, the max TS wil be less than 10 seconds.

BTW, do you know if there is some reasoning behind the `max_adv_duration = NODE_ID_TIMEOUT / MAX(cnt, 6);`? It is a remnant from MshPRF1.0.1, so does this calculation fully consider that we now have proxy advertising that depends on the On-Demand Private GATT Proxy state (1-255 seconds)?

Reply:
> I thought that it was already stating that it applied to multi subnet scenario, but if it is unclear I can rephrase it. The main purpose of this note is to emphazies that in this particular test scenario the max TS is 10 seconds, so that the reader might get the timing/adv_count criteria in the test. For a test/usecase with more than 6 subnets, with active proxy adv, the max TS wil be less than 10 seconds.

Ah, ok, true, my mistake.

> BTW, do you know if there is some reasoning behind the max_adv_duration = NODE_ID_TIMEOUT / MAX(cnt, 6);? It is a remnant from MshPRF1.0.1, so does this calculation fully consider that we now have proxy advertising that depends on the On-Demand Private GATT Proxy state (1-255 seconds)? As an example, a device has two subnets, where the first subnet has ongoing active proxy adv work, and then Private Network Identity adv is triggered on the second subnet via solicitation. With the suggested implementation they will get a 10 sec timeslot each, meaning that if the On-Demand Private GATT Proxy state is set to lets say 17 seconds it will in reality only advertise for 10 seconds.

I don't know why it is split into at least 6 slices. Perhaps to alternate between subnets more often, otherwise, e.g. with 2 subnets, you need to wait for 30 seconds before seeing another subnet advertisements. Regarding the solicitation, as we discussed offline, the solicited subnet should get exclusive rights to the advertiser. Perhaps, the implementation should consider that the solicited subnet adv duration can be more than 60 seconds.

> Is there a good reason why the proxy adv internals should not be triggered more frequently, lets say with a 2 second interval. Of course it will cause some extra power and resource consumption, but is it substantial? Should we consider a Kconfig option for the max duration rather than a fixed one?

I think a GATT client may have less chance to connect to the device.